packages/eslint-plugin/src/rules/switch-exhaustiveness-check.ts

```diff
@@ -161,7 +181,7 @@ export default createRule<Options, MessageIds>({
 
       return {
         containsNonLiteralType,
-        defaultCase,
+        defaultCase: defaultCase ? defaultCase : getCommentDefaultCase(node),
```

Comment:
[Style] Nit: I'm surprised this isn't caught by https://typescript-eslint.io/rules/prefer-nullish-coalescing ...

```suggestion
        defaultCase: defaultCase ?? getCommentDefaultCase(node),
```

Reply:
https://github.com/typescript-eslint/typescript-eslint/issues/10470